### PR TITLE
Taras/feature/voip

### PR DESCRIPTION
**Pull request by Taraslysak**

### CODE REVIEW

File: src/matrix/index.ts [hunk 1]
```diff
@@ -21,18 +22,21 @@ export class MatrixService implements IMatrixService, IAccessTokenHolder {
...
   contacts: MatrixContactService;
+  voip: MatrixVoipService;
   session: typeof matrixSession;
 
   constructor(
     rooms = MatrixRoomsService,
     messages = MatrixMessageService,
     contacts = MatrixContactService,
     session = matrixSession,
+    voip = MatrixVoipService,
```
Comment (kipod) [inline src/matrix/index.ts:33]:
```typescript
    Rooms = MatrixRoomsService,
    Messages = MatrixMessageService,
    Contacts = MatrixContactService,
    Session = matrixSession,
    Voip = MatrixVoipService,
```
I very much recommend name types with a capital letter

File: src/services/InvitationManager/InvitationManager.test.ts [hunk 2]
```diff
@@ -96,7 +97,7 @@ describe('Invitation manager', () => {
     expect(mockNavigate).toBeCalledWith('Private Chat', {
       chatId: 'network_001',
     });
-    expect(mockStore.dispatch).toBeCalledTimes(8);
+    expect(mockStore.dispatch).toBeCalledTimes(9);
```
Comment (kipod) [inline src/services/InvitationManager/InvitationManager.test.ts:100]: magic value? Consider to declare as const

File: src/services/InvitationManager/InvitationManager.test.ts [hunk 3]
```diff
@@ -134,6 +135,7 @@ describe('Invitation manager', () => {
     });
     expect(savedChat.name).toBe('test name 2');
     expect(savedChat.chatType).toBe('group');
+    savedChat.members.forEach((member) => console.log(member.networkId));
```
Comment (kipod) [inline src/services/InvitationManager/InvitationManager.test.ts:138]: Is it debug code?

File: src/services/Voip/Voip.ts [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+          type: ActionTypes.UPDATE_VOIP_STATE,
+          payload: {
+            callType: CallTypes.pending,
+            opponentNetworkId: this.chatId,
+          },
+        });
+        InCallManager.stopRingtone();
+      }
+    } catch (err) {
+      console.log('Accept Error', err);
```
Comment (kipod) [inline src/services/Voip/Voip.ts:148]: console.warn ?

**Review by kipod — COMMENTED** [review]
Thank you for this great opportunity to make a code review for your code! :)

File: src/components/buttons/CallPrimaryButton/CallPrimaryButton.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import {TouchableOpacity} from 'react-native';
+import {PhoneCallIcon, PhoneHangUpIcon} from '../../../assets';
+import {VoipService} from '../../../services';
+import {Colors} from '../../../themes';
+import styles from './style';
+interface Props {
+  variant: 'accept' | 'hangUp';
+}
+
+const ICON_SIZE = '50';
```
[the remark this answers is not in the document]
Comment (Taraslysak) [reply src/components/buttons/CallPrimaryButton/CallPrimaryButton.tsx:11]: Had issues in past when such parameter is passed to Svg library component as number, and in the rest of the project such values are used as string, so I think this better stay as string

File: __mocks__/react-native-webrtc.js [hunk 6]
```diff
@@ -0,0 +1 @@
+// export default {};
```
[the remark this answers is not in the document]
Comment (Taraslysak) [reply __mocks__/react-native-webrtc.js:1]: Uncommented. In any case, the file is needed for tests to work

File: android/gradle.properties [hunk 7]
```diff
@@ -31,3 +33,6 @@ org.gradle.jvmargs=-Xmx4096m -XX:MaxPermSize=4096m -XX:+HeapDumpOnOutOfMemoryErr
 org.gradle.daemon=true
 org.gradle.parallel=true
 org.gradle.configureondemand=true
+
+# Bigfix to ract native webrtc
```
[the remark this answers is not in the document]
Comment (Taraslysak) [reply android/gradle.properties:37]: Made comment more clear

File: ios/Podfile [hunk 8]
```diff
@@ -1,7 +1,7 @@
 require_relative '../node_modules/react-native/scripts/react_native_pods'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 
-platform :ios, '10.0'
+platform :ios, '11.0'
```
[the remark this answers is not in the document]
Comment (Taraslysak) [reply ios/Podfile:4]: Done

File: src/services/InvitationManager/InvitationManager.test.ts (hunk 3 above)
Comment (Taraslysak) [reply src/services/InvitationManager/InvitationManager.test.ts:138]: Removed

File: src/services/Voip/Voip.ts (hunk 4 above)
Comment (Taraslysak) [reply src/services/Voip/Voip.ts:148]: Done

**Pull request merged**